Python/Product/Cookiecutter/Model/CookiecutterClient.cs

```diff
@@ -72,43 +77,76 @@ class CookiecutterClient : ICookiecutterClient {
+        // the venv.
+        // Therefore, we have to use the python intepreter to check the real path of the venv,
+        // in case it's been redirected.
+        private async Task<string> GetRealPath(string path) {
+
+            // The expected path is going to be passed into python as a string, which means backslashes have to be escaped.
+            var expectedPath = path.Replace("\\", "\\\\");
+            var command = $"import os; print(os.path.realpath('{expectedPath}'))";
             var output = ProcessOutput.Run(
                 _interpreter.InterpreterExecutablePath,
-                new[] { "-m", "venv", _envFolderPath },
+                new[] { "-c", command },
                 null,
                 null,
                 false,
                 _redirector
             );
-            await WaitForOutput(_interpreter.InterpreterExecutablePath, output);
+
+            var result = await WaitForOutput(_interpreter.InterpreterExecutablePath, output);
+            return result.StandardOutputLines.FirstOrDefault();
         }
 
         private async Task CreateVenvWithoutPipThenInstallPip() {
+            
+            // The venv is not guaranteed to be created where expected, see GetRealPath() for more information.
+
+            // Also, Python has a bug (https://bugs.python.org/issue45337) where it doesn't
+            // keep track of the real location of the redirected venv when creating a venv with pip installed.
+            // In this case, the call to python.exe will have an exit code of 106.
+
+            // Therefore, the workaround is the following:
+            // 1. Create the venv WITHOUT PIP every time
+            // 2. Run python and check os.path.realpath against the expected venv path
+            // 3. If the real path comes back different, that's the real venv path.
+            // 5. Install pip using python.exe and the real venv path.
+
             RemoveExistingVenv();
 
-            _redirector.WriteLine(Strings.InstallingCookiecutterCreateEnvWithoutPip.FormatUI(_envFolderPath));
+            // create the venv without pip installed
+            _redirector.WriteLine(Strings.InstallingCookiecutterCreateEnvWithoutPip.FormatUI(_expectedEnvFolderPath));
             var output = ProcessOutput.Run(
                 _interpreter.InterpreterExecutablePath,
-                new[] { "-m", "venv", _envFolderPath, "--without-pip" },
+                new[] { "-m", "venv", _expectedEnvFolderPath, "--without-pip" },
                 null,
                 null,
                 false,
                 _redirector
             );
             await WaitForOutput(_interpreter.InterpreterExecutablePath, output);
 
+            // If we get here, the environment was created successfully.
+            // Update the envFolderPath, in case it's been redirected.
+            _envFolderPath = Task.Run(() => GetRealPath(_expectedEnvFolderPath)).Result;
```

Comment:
We can `await` here, so should definitely do that rather than blocking on the task.